client/sites/overview/components/migration-overview/components/cancel-difm-migration/index.tsx

```diff
@@ -3,13 +3,90 @@ import { Modal, Button } from '@wordpress/components';
 import { useTranslate } from 'i18n-calypso';
 import { useState } from 'react';
 import { useDispatch } from 'react-redux';
+import { useBlogStickersQuery } from 'calypso/blocks/blog-stickers/use-blog-stickers-query';
 import { LoadingBar } from 'calypso/components/loading-bar';
 import wp from 'calypso/lib/wp';
 import { successNotice, errorNotice } from 'calypso/state/notices/actions';
 
+const CancelMigrationButton = ( {
+	isMigrationInProgress,
+	onClick,
+}: {
+	isMigrationInProgress: boolean;
+	onClick: () => void;
+} ) => {
+	const translate = useTranslate();
+
+	const cancelMigrationButtonText = ! isMigrationInProgress
+		? translate( 'Cancel migration' )
+		: translate( 'Request migration cancellation' );
+
+	return (
+		<Button
+			variant="link"
+			onClick={ ( e: React.MouseEvent< HTMLButtonElement > ) => {
+				e.preventDefault();
+				onClick();
+			} }
+		>
+			{ cancelMigrationButtonText }
+		</Button>
+	);
+};
+
+const CancelMigrationModal = ( {
+	closeModal,
+	handleCancelMigration,
+	siteId,
+	isMigrationInProgress,
+}: {
+	closeModal: () => void;
+	handleCancelMigration: ( siteId: number ) => void;
+	siteId: number;
+	isMigrationInProgress: boolean;
+} ) => {
+	const translate = useTranslate();
+
+	const modalTitle = ! isMigrationInProgress
+		? translate( 'Cancel migration' )
+		: translate( 'Request migration cancellation' );
+
+	const cancelButtonText = ! isMigrationInProgress
+		? translate( 'Cancel migration' )
+		: translate( 'Send request' );
+
+	const modalContent = ! isMigrationInProgress
+		? translate(
+				'If you cancel now, our Happiness Engineers will be notified that you’ve chosen not to move your site to WordPress.com, and your current site will remain exactly as it is.'
+		  )
+		: translate(
+				"Since your migration is already underway, you'll need to send us a cancellation request. If you cancel now, you'll lose all your progress."
+		  );
+
+	return (
+		<Modal
+			className="migration-started-difm__cancel-dialog"
+			title={ modalTitle }
+			onRequestClose={ closeModal }
+			size="medium"
+		>
+			<p>{ modalContent }</p>
+			<div className="migration-started-difm__cancel-dialog-buttons">
+				<Button key="cancel" variant="secondary" onClick={ () => closeModal() }>
+					{ translate( "Don't cancel migration" ) }
+				</Button>
+				<Button key="send" variant="primary" onClick={ () => handleCancelMigration( siteId ) }>
+					{ cancelButtonText }
+				</Button>
```

Comment:
Would it be possible to use the CancelMigrationButton component to avoid repeating(almost) the button text?

Reply:
That's a good thought. I'll see if I can clean that up.